File: PJS5_POC_Cpp/Sources/Weapon/ProjectileWeapon/ProjectileWeapon.cpp
#include "ProjectileWeapon.h"

namespace WEAPON
{
	PROJECTILE_WEAPON::PROJECTILE_WEAPON(const int iId, const std::string& sName, const int iArmor, const int iLifePoint, const int iDamage, const int iPowerConsumption, const int iAccuracy, const int iMinAccuracy, const int iAmmo)
		: IWEAPON::IWEAPON(iId, sName, iArmor, iLifePoint, iDamage, iPowerConsumption, iAccuracy, iMinAccuracy, WEAPON_TYPE::PROJECTILE_WEAPON),
		_iAmmo(iAmmo)
	{
	}
}

File: PJS5_CSharp/PJS5_CSharp/Sources/Weapon/ProjectileWeapon/ProjectileWeapon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WEAPON;

namespace PJS5_CSharp.Sources.Weapon.ProjectileWeapon
{
    public class PROJECTILE_WEAPON : IWeapon
    {
        private int _iAmmo = -1;

        public PROJECTILE_WEAPON(int iId, string sName, int iArmor, int iLifePoint, int iDamage, int iPowerConsumption, int iAccuracy, int iMinAccuracy, int iAmmo)
            : base(iId, sName, iArmor, iLifePoint, iDamage, iPowerConsumption, iAccuracy, iMinAccuracy, WEAPON_TYPE.PROJECTILE_WEAPON)
        {
            _iAmmo = iAmmo;
        }

        public override int GetSpecificity()
        {
            return _iAmmo;
        }

        public void RemoveAmmo()
        {
            _iAmmo--;
        }
    }
}
